rspace/src/test/scala/coop/rchain/rspace/PointerBlockTests.scala

```diff
@@ -0,0 +1,103 @@
+package coop.rchain.rspace
+
+import coop.rchain.rspace.PointerBlock._
+import coop.rchain.rspace.test.ArbitraryInstances._
+import coop.rchain.rspace.test.roundTripCodec
+import coop.rchain.shared.AttemptOps._
+import org.scalacheck.Prop
+import org.scalactic.anyvals.PosInt
+import org.scalatest.prop.{Checkers, Configuration}
+import org.scalatest.{FlatSpec, Matchers}
+import scodec.bits.BitVector
+import scodec.{Codec, DecodeResult}
+
+class PointerBlockTests extends FlatSpec with Matchers with Checkers with Configuration {
+
+  implicit override val generatorDrivenConfig: PropertyCheckConfiguration =
+    PropertyCheckConfiguration(minSuccessful = PosInt(1000))
+
+  "A PointerBlock" should "be the same when round-tripped with scodec" in {
+
+    val propRoundTripCodec: Prop = Prop.forAll { (pb: PointerBlock) =>
+      roundTripCodec[PointerBlock](pb)
+        .map((value: DecodeResult[PointerBlock]) => value.value == pb)
+        .getOrElse(default = false)
+    }
+
+    check(propRoundTripCodec)
+  }
+
+  val emptyPb: PointerBlock = PointerBlock.create()
+
+  val emptyPbHash: Blake2b256Hash =
+    implicitly[Codec[PointerBlock]]
+      .encode(emptyPb)
+      .map((vector: BitVector) => Blake2b256Hash.create(vector.toByteArray))
+      .get
+
+  val helloHash: Blake2b256Hash = Blake2b256Hash.create("hello".getBytes())
+
+  val fullPb: PointerBlock = PointerBlock(Vector.fill(256)(Some(helloHash)))
+
+  "An empty PointerBlock" should "have the expected hash" in {
+
+    val expected = "2b69702a889248a4d6620475a105dccd5e0d4230aca8a492aaf6510e55d55b02"
+
+    emptyPbHash.bytes.toHex shouldBe expected
+  }
+
+  "A serialized empty PointerBlock" should "have the expected length" in {
+
+    val attemptedActual = implicitly[Codec[PointerBlock]]
+      .encode(emptyPb)
+      .map((vector: BitVector) => vector.bytes.length)
+
+    attemptedActual.get shouldBe PointerBlock.length
+  }
+
+  "A serialized full PointerBlock" should "have the expected length" in {
+
+    val attemptedActual = implicitly[Codec[PointerBlock]]
+      .encode(fullPb)
+      .map((vector: BitVector) => vector.bytes.length)
+
+    attemptedActual.get shouldBe (PointerBlock.length * (Blake2b256Hash.length + 1))
```

Comment:
Why `+1` ? Also, I should have asked this question in the previous PR, how is it working that you can [specify length](https://github.com/henrytill/rchain/blob/2f39f1608cbc5e907a1aae227caea4e21aaa9ad1/rspace/src/main/scala/coop/rchain/rspace/Blake2b256Hash.scala#L44) of the byte array no matter the initial length of your `Blake2b256Hash` [bytes](https://github.com/henrytill/rchain/blob/2f39f1608cbc5e907a1aae227caea4e21aaa9ad1/rspace/src/main/scala/coop/rchain/rspace/Blake2b256Hash.scala#L15) ?